src/Umbraco.Core/Persistence/Repositories/Implement/RedirectUrlRepository.cs

```diff
@@ -218,7 +218,21 @@ public IRedirectUrl GetMostRecentUrl(string url, string culture)
                 .OrderByDescending<RedirectUrlDto>(x => x.CreateDateUtc);
             var dtos = Database.Fetch<RedirectUrlDto>(sql);
             var dto = dtos.FirstOrDefault();
+
+            if (dto == null)
+                dto = GetMostRecentUrlWithNoCulture(url);
+
             return dto == null ? null : Map(dto);
         }
+
+        private RedirectUrlDto GetMostRecentUrlWithNoCulture(string url)
+        {
+            var urlHash = url.GenerateHash<SHA1>();
+            var sql = GetBaseQuery(false)
+                .Where<RedirectUrlDto>(x => x.Url == url && x.UrlHash == urlHash && x.Culture == string.Empty)
```

Comment:
Is there any reason why we can't 'just' have the single query above check `x.Culture == culture.ToLower() || x.Culture == string.Empty)` ? Would be better to have less queries if possible.

Reply:
I've combined it into a single DB query, but this means there is filtering on the Linq expression (FirstOrDefault) now. I hope this is okay?